src/Lumi/Components/Color.purs

```diff
@@ -114,3 +117,23 @@ colorNames =
   , transparent: ColorName "transparent"
   }
 
+shade ::
+  { hue :: Color, white :: Color, black :: Color } ->
+  { black :: Color
+  , grey1 :: Color
+  , grey2 :: Color
+  , hue :: Color
+  , hueDarker :: Color
+  , hueDarkest :: Color
+  , hueDisabled :: Color
+  , white :: Color
+  }
+shade { hue, white, black } =
```

Comment:
This seems quite specific to the `Button` and `ButtonGroup` components. Maybe we should call it `buttonShades` or move it somewhere else?

Reply:
Oh haha, I moved it here because it didn't seem buttons specific. Like part of the theme is all our current colors/hues, and this sort of expands each one into a wider range of related colors. I also wondered if it'd be better to move this function into the theme so it can be replaced.

Reply:
Hm I see… I can see the use for `hueDarker`, `hueDarkest`, and `hueDisabled, but maybe we should grab `grey1` and `grey2` from the theme itself? We alredy have `black1` and `black2`.

(this is totally non-blocking btw, just pure nitpicking)

Reply:
True. We should probably expand all the colors instead of just black if we can do it algorithmically, or leave the theme colors as-is and restrict button color to those predefined ones. The whole theme idea could use some work, like making it available to design and having more things use it so it doesn't look silly as values change, dark mode, etc.